lib/publiccloud/ssh_interactive.pm

```diff
@@ -125,4 +125,41 @@ sub select_host_console {
     assert_script_run("true", fail_message => "host console is broken");    # basic health check
 }
 
+sub prepare_ssh_tunnel {
+    my $instance = shift;
+
+    # configure ssh client
+    assert_script_run('curl -sf ' . data_url('publiccloud/ssh_config') . ' -o ~/.ssh/config');
+
+    # Create the ssh alias
+    assert_script_run(sprintf(q(echo -e 'Host sut\n  Hostname %s' >> ~/.ssh/config), $instance->public_ip));
+
+    # Copy SSH settings also for normal user
+    assert_script_run("install -o $testapi::username -g users -m 0700 -dD /home/$testapi::username/.ssh");
+    assert_script_run("install -o $testapi::username -g users -m 0600 ~/.ssh/* /home/$testapi::username/.ssh/");
+
+    # Skip setting root password for img_proof, because it expects the root password to NOT be set
+    $instance->ssh_assert_script_run(qq(echo -e "$testapi::password\\n$testapi::password" | sudo passwd root));
+
+    # Permit root passwordless login over SSH
+    $instance->ssh_assert_script_run('sudo sed -i "s/PermitRootLogin no/PermitRootLogin prohibit-password/g" /etc/ssh/sshd_config');
+    $instance->ssh_assert_script_run('sudo systemctl reload sshd');
+
+    # Copy SSH settings for remote root
+    $instance->ssh_assert_script_run('sudo install -o root -g root -m 0700 -dD /root/.ssh');
+    $instance->ssh_assert_script_run(sprintf("sudo install -o root -g root -m 0644 /home/%s/.ssh/authorized_keys /root/.ssh/", $instance->{username}));
+
+    # Create remote user and set him a password
+    $instance->ssh_assert_script_run("test -d /home/$testapi::username || sudo useradd -m $testapi::username");
```

Comment:
I assume it was not intentional , you simply haven't re-based for quite some time so you dropping here https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/c5b5e1c227ac9908b08439d73345fb057dc62ca2 which was merged recently

Reply:
> I assume it was not intentional , you simply haven't re-based for quite some time so you dropping here [c5b5e1c](https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/c5b5e1c227ac9908b08439d73345fb057dc62ca2) which was merged recently

Thank you so much for this. Done.